### PR TITLE
Fire backup operations even if the original operation returns no response

### LINKED ISSUE #15175: XA compatibility regression between 3.9 and 3.10 - Commit of prepared Xid on second XAResource fails
XA compatibility regression between 3.9 and 3.10
A commit of prepared Xid on second XAResource fails.
See also #2696

### PR DESCRIPTION
**Pull request by kwart**
Fixes #15175 

We failed to fire a backup operation if the original operation returned no response.

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/Operation.java [hunk 1]
```diff
@@ -182,7 +182,7 @@ public CallStatus call() throws Exception {
         }
 
         run();
-        return returnsResponse() ? DONE_RESPONSE : DONE_VOID;
```
Comment (pveentjer) [inline hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/Operation.java]: I do not understand this change.

We change this for all operations, even ones that don't have a backup. 

Also the DONE_VOID is doing exactly what is needed; the operation is complete and there is no response.

Comment (kwart) [reply hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/Operation.java]: Let me explain it.

The `OperationsRunnerImpl` based on the returned value decides if it even tries to send backup or not:

https://github.com/hazelcast/hazelcast/blob/66387b8dbdf0febc129ef30556ac93c8781f298c/hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/OperationRunnerImpl.java#L220-L247

We have some operations which don't return response but they still are `BackupAware` ones. We need to check if we should fire the backup operations for them.

The reported issue is caused by not sending backup operation for `FinalizeRemoteTransactionOperation` which returns `false` from the `returnsResponse()` method:

https://github.com/hazelcast/hazelcast/blob/66387b8dbdf0febc129ef30556ac93c8781f298c/hazelcast/src/main/java/com/hazelcast/transaction/impl/xa/operations/FinalizeRemoteTransactionOperation.java#L82-L100

Comment (mdogan) [reply hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/Operation.java]: Actually only difference between `DONE_VOID`  and `DONE_VOID_BACKUP` is, the latter sends backups if needed. `backupHandler.sendBackups(op)` is no-op if operation is not backup-aware or will not send any backups. 
After a second thought I think we can get rid of `DONE_VOID_BACKUP` completely and change behaviour of `DONE_VOID` to send backups when needed.

Comment (mmedenjak) [reply hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/Operation.java]: Discussed this a bit with @pveentjer , the problem is we're conflating concerns - one return ordinal determines two things: should we send a response to the caller and secondly - should we make backups or not.

The ideal solution IMHO would be that operations can decide this with two methods: `returnsResponse` and `sendsBackups`. But that might be a larger change with questionable benefits as I don't know how many operations need some combinations.

If we can do what Mehmet suggested (get rid of `DONE_VOID_BACKUP` completely and change behaviour of DONE_VOID to send backups when needed) without adversely impacting the performance of operations that previously used `DONE_VOID` and didn't need to send backups, I'm for it.

**Comment by pveentjer** [comment]
With the following PR https://github.com/hazelcast/hazelcast/pull/16289 this PRD should not be needed (although I would like to have the extra tests)

**Comment by kwart** [comment]
This PR was rebased to the @pveentjer 's  branch (#16289). So it only adds the regression test in the second commit.

**Comment by kwart** [comment]
Unrelated test failure in PR builder. Merging.

**Pull request merged**